### PR TITLE
Attach position to errors in nback

### PR DESCRIPTION
**Pull request by heshanpadmasiri**
## Purpose
Attach position to errors in nback
Related to #677 

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: compiler/modules/nback/scaffold.bal [hunk 1]
```diff
@@ -262,8 +262,8 @@ class Scaffold {
         }
     }
 
-    function unimplementedErr(err:Message message) returns err:Unimplemented {
-        err:Location loc = err:location(self.file);
+    function unimplementedErr(err:Message message, err:Position pos) returns err:Unimplemented {
+        err:Location loc = err:location(self.file, pos);
```
Comment (jclark) [inline compiler/modules/nback/scaffold.bal]: I would inline this.

File: compiler/modules/nback/build.bal [hunk 2]
```diff
@@ -222,7 +222,7 @@ function buildConvertRepr(llvm:Builder builder, Scaffold scaffold, Repr sourceRe
         }
     }
     // this shouldn't ever happen I think
-    return scaffold.unimplementedErr("unimplemented conversion required");
+    return scaffold.unimplementedErr("unimplemented conversion required", pos);
```
Comment (jclark) [inline compiler/modules/nback/build.bal]: I think it's easier to turn this into

```
panic err:impossible("unexpected ...");
```

File: compiler/modules/nback/scaffold.bal (hunk 1 above)
Comment (heshanpadmasiri) [reply compiler/modules/nback/scaffold.bal]: fixed

**Review by jclark — CHANGES_REQUESTED** [review]

File: compiler/modules/nback/build.bal (hunk 2 above)
Comment (jclark) [reply compiler/modules/nback/build.bal]: And then we don't need to bother with passing around the position.

Comment (heshanpadmasiri) [reply compiler/modules/nback/build.bal]: fixed

**Review by jclark — APPROVED** [review]

**Pull request merged**